util/hint/hint_processor.go

```diff
@@ -435,6 +440,26 @@ func (p *BlockHintProcessor) getBlockOffset(blockName model.CIStr) int {
 	return -1
 }
 
+// ParseOffsetFromBlockName parse block offset from blockName
+func (p *BlockHintProcessor) ParseOffsetFromBlockName(blockName model.CIStr, curLevel int) int {
+	if p.QbNameMap != nil {
+		level, ok := p.QbNameMap[blockName.L]
+		if ok {
+			return level
+		}
+	}
+
+	if strings.HasPrefix(blockName.L, defaultSelectBlockPrefix) {
+		suffix := blockName.L[len(defaultSelectBlockPrefix):]
+		level, err := strconv.ParseInt(suffix, 10, 64)
+		if err != nil {
+			return -1
+		}
+		return int(level)
```

Comment:
Should we store the `level` to `p.QbNameMap`?

Reply:
p.QbNameMap is used to Map from query block name to select stmt offset.in this time there are not exists qb_name hint that used to ref query block